testing/integration/upgrade_test.go

```diff
@@ -332,3 +326,216 @@ func (s *UpgradeStandaloneElasticAgent) TestUpgradeStandaloneElasticAgentToSnaps
+	s.Require().NoError(err)
+
+	s.T().Log("Modify /etc/hosts to simulate transient network error")
+	cmd := exec.Command("sed",
+		"-i.bak",
+		"s/localhost/localhost artifacts.elastic.co artifacts-api.elastic.co/g",
+		"/etc/hosts",
+	)
+	s.T().Log("/etc/hosts modify command: ", cmd.String())
+
+	output, err = cmd.CombinedOutput()
+	if err != nil {
+		s.T().Log(string(output))
+	}
+	s.Require().NoError(err)
+
+	// Ensure that /etc/hosts is modified
+	s.Eventually(func() bool {
+		cmd := exec.Command("grep",
+			"artifacts",
+			"/etc/hosts",
+		)
+		s.T().Log("Check /etc/hosts command: ", cmd.String())
+
+		// We don't check the error as grep will return non-zero exit code when
+		// it doesn't find any matches, which could happen the first couple of
+		// times it searches /etc/hosts.
+		output, _ := cmd.CombinedOutput()
+		outputStr := strings.TrimSpace(string(output))
+
+		return outputStr != ""
+	}, 10*time.Second, 1*time.Second)
+
+	s.isEtcHostsModified = true
+	defer s.restoreEtcHosts()
+
+	s.T().Log("Start the Agent upgrade")
+	var toVersion = s.toVersion.String()
+	var wg sync.WaitGroup
+	go func() {
+		wg.Add(1)
+
+		err := s.upgradeAgent(ctx, toVersion)
+
+		wg.Done()
+		s.Require().NoError(err)
+	}()
+
+	s.T().Log("Check Agent logs for at least two retry messages")
+	agentDirName := fmt.Sprintf("elastic-agent-%s", release.TrimCommit(currentVersion.Daemon.Commit))
+	logsPath := filepath.Join(paths.DefaultBasePath, "Elastic", "Agent", "data", agentDirName, "logs")
+	s.Eventually(func() bool {
+		cmd := exec.Command("grep",
+			"download.*retrying",
+			"--recursive",
+			"--include", "*.ndjson",
+			logsPath,
+		)
+		s.T().Log("Find logs command: ", cmd.String())
+
+		// We don't check the error as grep will return non-zero exit code when
+		// it doesn't find any matches, which could happen the first couple of
+		// times it searches the Elastic Agent logs.
+		output, _ := cmd.CombinedOutput()
+		outputStr := strings.TrimSpace(string(output))
+
+		outputLines := strings.Split(outputStr, "\n")
+		s.T().Log(outputLines)
+		s.T().Log("Number of retry messages: ", len(outputLines))
+		return len(outputLines) >= 2
+	}, 2*time.Minute, 20*time.Second)
+
+	s.T().Log("Restore /etc/hosts so upgrade can proceed")
+	s.restoreEtcHosts()
+
+	// Wait for upgrade command to finish executing
+	s.T().Log("Waiting for upgrade to finish")
+	wg.Wait()
+
+	s.T().Log("Check Agent version to ensure upgrade is successful")
+	currentVersion, err = s.getVersion(ctx)
+	s.Require().NoError(err)
+	s.Require().Equal(toVersion, currentVersion.Binary.Version)
+	s.Require().Equal(toVersion, currentVersion.Daemon.Version)
+}
+
+func (s *StandaloneUpgradeRetryDownloadTestSuite) getVersion(ctx context.Context) (*versionOutput, error) {
+	var currentVersion versionOutput
+	var err error
+
+	s.Eventually(func() bool {
+		args := []string{"version", "--yaml"}
+		var output []byte
+		output, err = s.agentFixture.Exec(ctx, args)
+
+		if err != nil {
```

Comment:
[Go convention and consistency]
Please...
```suggestion
		output, err = s.agentFixture.Exec(ctx, args)
		if err != nil {
```

Reply:
Didn't mean to ignore this suggestion.  The PR was set to auto-merge so it got merged immediately upon seeing the approval.  Implemented this suggestion in https://github.com/elastic/elastic-agent/pull/2827.